Mathlib/Data/Finsupp/Basic.lean

```diff
@@ -1471,8 +1471,10 @@ end
 
 section
 
-instance smulZeroClass [Zero M] [SMulZeroClass R M] : SMulZeroClass R (α →₀ M) where
+instance smul [Zero M] [SMulZeroClass R M] : SMul R (α →₀ M) where
   smul a v := v.mapRange ((· • ·) a) (smul_zero _)
+
+instance smulZeroClass [Zero M] [SMulZeroClass R M] : SMulZeroClass R (α →₀ M) where
   smul_zero a := by
     ext
     apply smul_zero
```

Comment:
I would be surprised (and a bit sad) if this helps, but I guess I was surprised by the Kaehler PR working too...

Did you ever test this separately from the other two changes, to get a feel for which is most significant?

Reply:
I'm in the process of doing some extensive testing now. Will report back.